lib/src/main/kotlin/org/didcommx/didcomm/exceptions/Exceptions.kt

```diff
@@ -9,11 +12,42 @@ package org.didcommx.didcomm.exceptions
 open class DIDCommException(message: String, cause: Throwable? = null) : Throwable(message, cause)
 
 /**
- * This exception SHOULD be raised if secret can not be converted to JWK format.
+ * This exception SHOULD be raised if verification method type is not supported.
+ *
+ * @param type The verification method type.
+ */
+class UnsupportedVerificationMethodTypeException(type: VerificationMethodType) :
```

Comment:
I think we have too many similar Unsupported exceptions. Most probably it doesn't make sense for the user to catch so many, and it's hard to say what exception to catch.
I would make just one DIDCommUnsupportedException, and put the details about what is unsupported into the message itself. We can put there an optional error code as well.

Reply:
Added `DIDCommUnsupportedValueException` base class.